include/EVT/dev/button.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef _EVT_BUTTON_
+#define _EVT_BUTTON_
+
+#include <EVT/io/GPIO.hpp>
+#include <stdint.h>
+
+namespace EVT::core::DEV {
+// Forward declarations:
+// The different pins are hardware specific. Forward declaration to allow
+// at compilation time the decision of which pins should be used.
+
+class Button {
+public:
+    /**
+         * Enum to handle the button trigger states
+         *
+         */
+    enum class LogicLevelOnPress {
+        LOW = 0u,
+        HIGH = 1u,
+    };
+
+    /**
+         * Create an instance of the LED based on the given GPIO pin.
+         *
+         * @param gpio[in] GPIO pin
+         * @param logicLevelOnPress[in] Represents if the button is active high or active low
+         */
+    Button(EVT::core::IO::GPIO& gpio, Button::LogicLevelOnPress logicLevelOnPress);
+
+    /**
+         * Reads the state of the button
+         *
+         * @return The button state.
+         */
+    IO::GPIO::State readButton();
```

Comment:
It may be overly pedantic, but I'm not sure readButton() should return a `IO::GPIO::State`.  We are looking for a value of pressed or not pressed here and returning a GPIO value of HIGH or LOW may cause some confusion.